tests/smoke_tests/monitors_test/syslog_test.py

```diff
@@ -66,28 +66,27 @@ def _test(python_version):
     runner.start(executable=python_version)
     time.sleep(1)
     reader = LogReader(runner.syslog_log_path)
-
-    reader.start(wait_for_data=False)
+    agent_log_reader = AgentLogReader(runner.agent_log_file_path)
 
     os.system("logger message1 --port {0} --udp --server 127.0.0.1".format(UDP_PORT))
 
-    last_line = reader.wait_for_new_line()
+    last_line = reader.wait_for_next_line()
 
     assert "message1" in last_line
 
     os.system("logger message2 --port {0} --tcp --server 127.0.0.1".format(TCP_PORT))
 
-    last_line = reader.wait_for_new_line()
+    last_line = reader.wait_for_next_line()
 
     assert "message2" in last_line
 
     os.system("logger message3 --port {0} --udp --server 127.0.0.1".format(UDP_PORT))
 
-    last_line = reader.wait_for_new_line()
+    last_line = reader.wait_for_next_line()
 
     assert "message3" in last_line
 
-    runner.stop()
+    agent_log_reader.go_to_end()
```

Comment:
This works for now, but since we always need to call this ``go_to_end()`` method it may make sense in the future to implement ContextManager which always calls that method on ``__exit__``.

Reply:
Good advice. Added reminder for that.